src/libsyntax/ast.rs

```diff
@@ -1746,13 +1730,21 @@ impl StructFieldKind {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+pub enum VariantKind {
+    Dict,
```

Comment:
Aside from fixing bugs, decoupling structure kinds from field kinds can be seen as laying the ground for this model https://github.com/rust-lang/rfcs/pull/218#issuecomment-76393168